### ISSUE TITLE
Yank recent dask (distributed) releases?

### ISSUE THREAD

**Issue by fjetter** [dask/community#238]
We've observed a bunch of severe stability issues with the most recent versions.

It appears that the January version `2022.1.1` is the most recent stable version. Should we [yank](https://pypi.org/help/#yanked) the subsequent releases? Is there an equivalent for conda we can do?

- https://github.com/dask/distributed/issues/6110
- https://github.com/dask/distributed/issues/5951

**Comment by fjetter** [comment]
Obviously, we also shouldn't release anymore until we're reasonably confident that the new versions are actually stable.

**Comment by martindurant** [comment]
Can you please link any issues about stability, for reference. Is it just `distributed` we are talking about?

**Comment by gjoseph92** [comment]
- https://github.com/dask/distributed/issues/5951
-  https://github.com/dask/distributed/issues/5879
- https://github.com/dask/distributed/issues/6110
- https://github.com/dask/distributed/issues/5635 (haven't looked in a while, but probably not fixed. This is pre-2021.1.1 though.)
- Another one with different symptoms we don't have a public issue for yet (tasks are processing on scheduler, but all in memory on workers)
- https://github.com/dask/distributed/issues/5960 (not a deadlock, but somewhat serious for memory performance in certain cases)

**Comment by hayesgb** [comment]
I vote yes.

**Comment by jakirkham** [comment]
There's at least one fix in flight ( https://github.com/dask/distributed/pull/6112 ). Would suggest holding off here for the moment.

**Comment by gjoseph92** [comment]
Whether we will have a new release that fixes some issues isn't relevant to whether we yank existing releases that are known to be unstable.

**Comment by jakirkham** [comment]
This will impact deployments that have versions constraints that are tied to the yanked versions. RAPIDS would be negatively impacted for example.

**Comment by martindurant** [comment]
For a PyPI "yank", you can still get a specific version when explicitly specified, but it isn't included in a free install. That's something like putting a label on a conda package. You can also change the conda-forge metadata via PR quite easily, but I don't know how to get the exact same behaviour.

**Comment by mrocklin** [comment]
An alternative is to see what we can resolve in the next week, and plan to do a release on Friday.  

I plan to focus on fixing these issues this week regardless.

**Comment by mrocklin** [comment]
First PR (mostly just doing what Gabe and Florian suggested) is up here: https://github.com/dask/distributed/pull/6112

**Comment by gjoseph92** [comment]
> but were not considered release blockers at the time

Because they weren't known to be issues at the time (inadequate testing). If any one of them had been known to cause a deadlock, they would have been a release blocker. We just only recognize this in hindsight.

**Comment by quasiben** [comment]
Yanking (I think) 4 releases seems fairly extreme, I think that's what is being proposed here. As John said, RAPIDS just released 22.04 which relies on [2022.03.0](https://github.com/rapidsai/cudf/blob/branch-22.04/conda/recipes/dask-cudf/meta.yaml#L30) and yanking this out would be very disruptive even though there are known bugs.

**Comment by mrocklin** [comment]
Second PR (around work stealing) https://github.com/dask/distributed/pull/6115

This isn't actually about deadlocks, but it was easy to investigate.

**Comment by martindurant** [comment]
See https://github.com/dask/distributed/issues/6110

**Comment by mrocklin** [comment]
Hi All, weighing in here.  I'll include a quick status update, longer-term plans, and some general thoughts here.

## Quick status update

The latest release pushed out last Friday identified and resolved two known deadlocks.  We've identified a third in the issue that Martin just pointed to and have a fix for it in a PR (although that issue seems a bit more fringe).  That will go out next Friday.  

I encourage everyone to try latest release.

## Personal plans

I'm shifting my time away from adminsitrative work to be spending around 50% of my time on dask.distributed.  This has been going on for the last couple of weeks now.  I plan to keep this up for the next quarter.

Aside from this topic there is a lot of good work to do.  There are many things to clean up.  I'm having a good time and I encourage a positive outlook.  I apologize to anyone as I deprecate your favorite old feature :) 

## Thoughts on yanking

I'm open to yanking releases, but it's not my first choice.  I would rather we focus on fixing things and getting out a solid release than on retreating.  I think that with some focused effort we can resolve these problems in a couple more weeks (recent progress has been good, thanks everyone, let's keep it up).

I don't think that there is a very solid platform to retreat to anyway, and I don't want us to get into that habit, at least not when considering going far back in history.

## LTS

I like Dask's rapid and regular release cycle.  I think innovation is still more valuable to Dask than API stability, at least for the community development cycle.

`<heads-up, about to mention coiled things>`

However, I also appreciate the desire for a more steady base, especially in a more enterprise or institutional setting.  FWIW Coiled needed something like this for our customers (we need to have a thing to target for support components of contracts) and we published a decently simple meta-package at https://github.com/coiled/coiled-runtime .  There's also a nascent effort there to do larger scale integration testing.  We've found this metapackage useful and plan to maintain it.  It's free to use (on conda-forge) although we don't promise to provide free advice/support around it.  That might provide a bit more of a guided experience to folks looking for something that will be stable for a longer period of time.

## Release candidates

I'm fine with that if folks would actually opt-in to test and provide feedback.  My sense is that most of the larger downstream packages have us in an upstream CI build.  This seems to be as-or-more effective than what I've seen of release candidates in the PyData stack, but I'm totally open to this if folks want to start engaging.  

Question here: who would actively try things out and report back feedback?  If yes, what lead time would you need?
